docs/js-api.rst

```diff
@@ -0,0 +1,99 @@
+.. _js-api:
+
+JavaScript API Reference
+========================
+
+The :ref:`futhark-wasm(1)` and :ref:`futhark-wasm-multicore(1)`
+compilers produce JavaScript wrapper code to allow JavaScript programs
+to invoke the generated WebAssembly code.  This chapter describes the
+API exposed by the wrapper.
+
+First a warning: **the JavaScript API is experimental**.  It may
+change incompatibly even in minor versions of the compiler.
+
+FutharkContext
+--------------
+FutharkContext is a class that contains information about the context
+and configuration from the C API. It has methods for invoking the Futhark
+entry points and creating FutharkArrays on the WebAssembly heap.
+
+
+.. js:function:: constructor()
+
+   Creates a new FutharkContext.
+
+.. js:function:: free()
+
+   Frees all memory created by the FutharkContext object. Should be called
+   when the FutharkContext is done being used.
```

Comment:
Say that it is an error to use any `FutharkArray` or `FutharkOpaque` value connected to this context after it has been freed.